test/example/generated/ExampleDialect.cpp.inc

```diff
@@ -322,25 +322,25 @@ return true;
 
     
       ::llvm::Value * StreamReduceOp::getPtr() const {
-        return getArgOperand(0);
+        return getArgOperand(static_cast<uint32_t>(ArgumentIndex::Ptr));
```

Comment:
Now that I see all these static casts, maybe it would be better to define a non-class enum, which does not need explicit conversion to uint32_t:
```c++
// Define a non-class enum, so that it can be used as index without casts, but is still accessed like a class enum
struct ArgumentIndex {
enum : uint32_t {
  Ptr = 0,
  Count = 1,
};
};
```

Reply:
+1 on the idea to get rid of the casts, however, I think we should also make sure the enum type is accessible so it can be stored without auto, maybe like this?

```c++
struct ArgumentIndex {
enum Enum : uint32_t {
   Ptr = 0,
   Cont = 1
};
};
```